s3fs/core.py

```diff
@@ -13,6 +17,21 @@
 
 logging.getLogger('boto3').setLevel(logging.WARNING)
 logging.getLogger('botocore').setLevel(logging.WARNING)
+S3_RETRYABLE_ERRORS = (
+    socket.timeout,
+    trans.ReadTimeoutError, trans.IncompleteReadError
+)
```

Comment:
Different versions of botocore seem to have this in different places or not at all.